### PR TITLE
Implement Auction House fees - no more free ride kids.

### PR DESCRIPTION
**Pull request by TeoTwawki**
**I finally got around to it.**

Tested. Displays correctly, deducts correct amount of gil at correct time.

Comes with settings, defaults are what both wiki's say the retail fees are.

### CODE REVIEW

File: src/map/packet_system.cpp [hunk 1]
```diff
@@ -2163,18 +2163,33 @@ void SmallPacket0x04E(map_session_data_t* session, CCharEntity* PChar, CBasicPac
             !(PItem->isSubType(ITEM_LOCKED)) &&
             !(PItem->getFlag() & ITEM_FLAG_NOAUCTION))
         {
-            if (quantity == 0 &&
-                (PItem->getStackSize() == 1 ||
-                PItem->getStackSize() != PItem->getQuantity()))
+            uint32 AUCTION_FEE = 0;
```
Comment (teschnei) [inline src/map/packet_system.cpp]: don't use caps for a local variable (caps are usually reserved for macros and enums)

Comment (TeoTwawki) [inline src/map/packet_system.cpp]: I'll change it, it was copy pasta from the previous AUCTION_FEE garbage

Comment (TeoTwawki) [inline src/map/packet_system.cpp]: p.s. this is why I asked before when doing the style guide for the repo, I forget these things

**Comment by TeoTwawki** [comment]
Amended.

**Pull request merged**